tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs

```diff
@@ -70,5 +78,10 @@ public void GeneratedEntity_Sqlite_CanSaveAndReadFields_AllTypes()
         testEntity.MoneyTestField!.Value.Amount.Should().Be(money);
         testEntity.MoneyTestField.Value.CurrencyCode.Should().Be(currencyCode);
         testEntity.CountryCode2TestField!.Value.Should().Be(countryCode2);
+
+        var actualAddress = JsonSerializer.Serialize(testEntity.StreetAddressTestField!.Value);
+        var expectedAddress = JsonSerializer.Serialize(addressItem);
+
+        actualAddress.Should().Be(expectedAddress);
```

Comment:
I think you can achieve this with FluentAssertions https://fluentassertions.com/objectgraphs/